src/viewer/App.tsx

```diff
@@ -376,7 +401,9 @@ export class AppStateStore {
         frame.binary = stringToBuffer(response.payloadData);
       }
       const connection = this.ensureConnection(requestId);
+      const rSocketFrame = tryDeserializeFrame(frame.payload);
       connection.frames.push(frame)
+      connection.index.add(frame.id, (rSocketFrame as any)?.data ?? frame.text ?? frame.payload)
```

Comment:
In my local tests this works for text frames and rsocket frames.
Is there any other thing I missed?